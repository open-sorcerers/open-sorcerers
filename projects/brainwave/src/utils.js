import path from "path"
import globby from "globby"
import { futurizeWithCancel, tacit } from "ensorcel"
import { fork as rawFork } from "fluture"
import { curryN, __ as $ } from "ramda"

export const globWithCancel = futurizeWithCancel($, 1, globby)
export const fork = tacit(3, rawFork)
export const relativePath = curryN(2, path.resolve)

export const wrap = curryN(2, (kk, vv) => ({ [kk]: vv }))

const isString = x => typeof x === "string"

export const stringInString = curryN(
  2,
  (xx, yy) =>
    isString(xx) && isString(yy) && (xx.includes(yy) || yy.includes(xx))
)
