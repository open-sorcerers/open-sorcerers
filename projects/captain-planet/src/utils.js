const merge = curry((a, b) => Object.assign({}, a, b))
const safeJSONParse = x => {
  try {
    return JSON.parse(x)
  } catch (e) {
    return {}
  }
}
const j2 = x => JSON.stringify(x, null, 2)
// remove null values
const smooth = filter(I)
)
// read a file in the Future
const read = x => F.encaseN2(fs.readFile)(x, "utf8")
// write a file in the Future
const write = curry(
  (fd, data) =>
    new F((bad, good) => {
      fs.writeFile(fd, data, "utf8", (e, f) => (e ? bad(e) : good(f)))
    })
)

// an essential tool for using with `ap`
const box = x => [x]
const isArray = Array.isArray
