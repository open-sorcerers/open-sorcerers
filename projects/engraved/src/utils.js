import { memoizeWith, identity as I } from "ramda"
export const memo = memoizeWith(I)
export const j2 = x => JSON.stringify(x, null, 2)
