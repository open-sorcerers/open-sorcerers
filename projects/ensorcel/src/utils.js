import { filter, identity as I } from "ramda"

export const freeze = Object.freeze

export const j2 = x => JSON.stringify(x, null, 2)

export const smooth = filter(I)

export const box = x => [x]
