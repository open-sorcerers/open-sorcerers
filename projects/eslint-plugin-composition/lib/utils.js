const { curry, reduce, pipe, equals } = require('ramda');
const NO_MATCH = 'CAMPER_NO_MATCH';
const unfound = equals(NO_MATCH);
const safePathWalk = reduce((last, next) =>
  last && last[next] ? last[next] : NO_MATCH
);
// eslint-disable-next-line
const typeIs = y => x => typeof x === y;
const isFunction = typeIs('function');
const safePathOr = curry((alt, steps, x) =>
  pipe(
    safePathWalk(x),
    y => (unfound(y) ? alt : y)
  )(steps)
);
const fromSchema = curry((or, key, x) =>
  safePathOr(or, ['options', 0, key], x)
);

const safePathEq = curry((compare, steps, x) =>
  pipe(
    safePathWalk(x),
    y => (unfound(y) ? false : isFunction(compare) ? compare(y) : y === compare)
  )(steps)
);
const safePathNotEq = curry((compare, steps, x) =>
  pipe(
    safePathWalk(x),
    y =>
      unfound(y) ? false : isFunction(compare) ? !compare(y) : y !== compare
  )(steps)
);

module.exports = {
  NO_MATCH,
  fromSchema,
  safePathEq,
  safePathNotEq,
  safePathOr
};
