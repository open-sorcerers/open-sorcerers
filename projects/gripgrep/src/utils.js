import { once, propOr, equals, curryN, length } from "ramda"
import { NIL } from "./constants"

export const box = x => [x]
export const orNil = propOr(NIL)
export const isNil = equals(NIL)
export const KÎ© = once(x => () => x)
export const wrap = curryN(2, (ab, x) =>
  length(ab) === 2 ? ab[0] + x + ab[1] : ab + x + ab
)
export const quote = wrap(`'`)
// eslint-disable-next-line valid-typeof
export const expectType = curryN(2, (type, xx) => typeof xx === type)
export const expectWhat = curryN(
  2,
  (expected, what) =>
    `Expected to be given ${expected} but instead received (${typeof what})`
)
export const yellUnless = curryN(3, (yell, expected, what) => {
  if (!expectType(expected, what)) {
    yell(expectWhat(expected, what))
    return false
  }
  return true
})
