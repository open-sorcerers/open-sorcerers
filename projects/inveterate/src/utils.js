export const freeze = Object.freeze

