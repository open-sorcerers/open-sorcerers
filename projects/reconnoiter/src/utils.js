import { flexeca } from "flexeca"
import { fork as rawFork } from "fluture"
import { sideEffect, trace } from "xtrace"
import {
  trim,
  propOr,
  replace,
  gt,
  join,
  head,
  last,
  curry,
  always as K,
  range,
  not,
  equals,
  ifElse,
  identity as I,
  chain,
  nth,
  filter,
  split,
  map,
  pipe,
  includes
} from "ramda"
import { tacit } from "ensorcel"
/* import { every } from "./cron" */

export const fork = tacit(2, rawFork)

const cwd = process.cwd()
export const netset = flexeca({ cwd }, "networksetup")

export const isOn = () =>
  pipe(netset, map(includes("On")))(["-getairportpower", "en0"])

export const ping = curry((times, what) =>
  pipe(
    flexeca({ cwd }, "ping"),
    map(
      pipe(
        split("\n"),
        filter(includes("packet loss")),
        propOr("", 0),
        split(","),
        last,
        trim,
        split(" "),
        head,
        replace("%", ""),
        parseFloat,
        // 1 = perfect connectivity, 0 = shit
        z => Math.abs(100 - z) / 100
      )
    )
  )(["-c" + times, what])
)

export const on = () =>
  pipe(
    netset,
    map(() => true)
  )(["-setairportpower", "en0", "on"])

export const off = () =>
  pipe(
    netset,
    map(() => true)
  )(["-setairportpower", "en0", "off"])

export const ipAddress = () =>
  pipe(
    netset,
    map(
      pipe(split("\n"), filter(includes("IP address")), nth(0), z =>
        z.substr(z.indexOf(":") + 2, Infinity)
      )
    )
  )(["-getinfo", "Wi-Fi"])

export const testNetwork = () =>
  pipe(
    netset,
    map(pipe(includes("not associated"), not))
  )(["-getairportnetwork", "en0"])

export const connectTo = net => () =>
  pipe(netset)(["-setairportnetwork", "en0", net])
