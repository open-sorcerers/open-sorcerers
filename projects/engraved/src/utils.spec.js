import { memo, j2 } from "./utils"

test("memo - default", () => {
  let count = 0
  const x = aa => {
    count += 1
    return aa * aa * aa * aa * aa
  }
  expect(x()).toEqual(x())
  expect(count).not.toEqual(1)
})
test("memo", () => {
  let count = 0
  const y = aa => {
    count += 1
    return aa * aa * aa * aa * aa
  }
  const x = memo(y)
  expect(x()).toEqual(x())
  expect(count).toEqual(1)
})
test("j2", () => {
  expect(j2({ a: { b: { c: [123] } } })).toMatchSnapshot()
})
