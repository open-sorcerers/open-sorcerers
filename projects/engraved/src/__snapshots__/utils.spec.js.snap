// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`j2 1`] = `
"{
  \\"a\\": {
    \\"b\\": {
      \\"c\\": [
        123
      ]
    }
  }
}"
`;
